drivers/tca6507/tca6507.c

```diff
@@ -0,0 +1,327 @@
+    regs[1] = regs[1] & ~(1UL << led);
+    regs[2] = regs[2] & ~(1UL << led);
+
+    if (i2c_write_regs(TCA6507_I2C, TCA6507_ADDR,
+                       TCA6507_SELECT0 | TCA6507_COMMAND_AUTOINC, &regs,
+                       sizeof(regs), 0) != 0) {
+        DEBUG("[ERROR] Cannot write command 'TCA6507_COMMAND_AUTOINC' to I2C.\n");
+        goto finish;
+    }
+    ret = TCA6507_OK;
+
+finish:
+    i2c_release(TCA6507_I2C);
+    return ret;
+}
+
+int tca6507_clear_all(const tca6507_t *dev)
+{
+    int ret = TCA6507_ERR_I2C;
+    uint8_t regs[3];
+
+    /* setup the i2c bus */
+    i2c_acquire(TCA6507_I2C);
+
+    regs[0] = 0;
+    regs[1] = 0;
+    regs[2] = 0;
+
+    if (i2c_write_regs(TCA6507_I2C, TCA6507_ADDR,
+                       TCA6507_SELECT0 | TCA6507_COMMAND_AUTOINC, &regs,
+                       sizeof(regs), 0) != 0) {
+        DEBUG("[ERROR] Cannot write command 'TCA6507_COMMAND_AUTOINC' to I2C.\n");
+        goto finish;
+    }
+    ret = TCA6507_OK;
+
+finish:
+    i2c_release(TCA6507_I2C);
+    return ret;
+}
+
+
+int tca6507_toggle_led(const tca6507_t *dev, uint8_t led, uint8_t bank)
+{
+    int ret = TCA6507_ERR_I2C;
+    uint8_t regs[3];
+
+    /* setup the i2c bus */
+    i2c_acquire(TCA6507_I2C);
+
+    if (i2c_read_regs(TCA6507_I2C, TCA6507_ADDR,
+                      TCA6507_SELECT0 | TCA6507_COMMAND_AUTOINC, &regs,
+                      sizeof(regs), 0) != 0) {
+        DEBUG("[ERROR] Cannot read TCA6507 from I2C.\n");
+        goto finish;
+    }
+
+    /* only work if led was is state 000 or 010/110 before. */
+    if (bank == 0) {
+        regs[0] = regs[0] ^ 0UL << led;
+        regs[1] = regs[1] ^ 1UL << led;
+        regs[2] = regs[2] ^ 0UL << led;
+    }
+    else {
+        regs[0] = regs[0] ^ 1UL << led;
+        regs[1] = regs[1] ^ 1UL << led;
+        regs[2] = regs[2] ^ 0UL << led;
+    }
+
+    if (i2c_write_regs(TCA6507_I2C, TCA6507_ADDR,
+                       TCA6507_SELECT0 | TCA6507_COMMAND_AUTOINC, &regs,
+                       sizeof(regs), 0) != 0) {
+        DEBUG("[ERROR] Cannot write command 'TCA6507_COMMAND_AUTOINC' to I2C.\n");
+        goto finish;
+    }
+    ret = TCA6507_OK;
+
+finish:
+    i2c_release(TCA6507_I2C);
+    return ret;
+}
+
+int tca6507_blink_led(const tca6507_t *dev, uint8_t led, uint8_t bank)
+{
+    int ret = TCA6507_ERR_I2C;
+    uint8_t regs[3];
+    uint8_t mask = 1UL << led;
+
+    /* setup the i2c bus */
+    i2c_acquire(TCA6507_I2C);
+
+    if (i2c_read_regs(TCA6507_I2C, TCA6507_ADDR,
+                      TCA6507_SELECT0 | TCA6507_COMMAND_AUTOINC, &regs,
+                      sizeof(regs), 0) != 0) {
+        DEBUG("[ERROR] Cannot read TCA6507 from I2C.\n");
+        goto finish;
+    }
+
+    if (bank == 0) {
+        regs[0] = regs[0] & ~mask;
+        regs[1] = regs[1] | mask;
+        regs[2] = regs[2] | mask;
+    }
+    else {
+        regs[0] = regs[0] | mask;
+        regs[1] = regs[1] | mask;
+        regs[2] = regs[2] | mask;
+    }
+
```

Comment:
A comment on these things would be nice - I was wondering whether you forgot a `& ~mask;` in the second case.

Reply:
The code seems correct.
I will add a datasheet extract to explain the content of register 0-2.